EIPS/eip-5773.md

```diff
@@ -0,0 +1,362 @@
+    ///  that the the active resource at index 1 of the active resource array
+    ///  has a priority of 1, index 2 has a priority of 3, and index 3 has a priority
+    ///  of 2. There is no validation on priority value input; out of order indexes
+    ///  must be handled by the frontend.
+    /// @dev The length of the priorities array MUST
+    ///  be equal to the present length of the active resources array.
+    /// @param tokenId the ID of the token of the resource priority to set
+    /// @param priorities An array of priorities to set.
+    function setPriority(uint256 tokenId, uint16[] memory priorities) external;
+
+    /// @notice Returns an array of uint64 identifiers from the active resources
+    ///  array for resource lookup.
+    /// @dev Each uint64 resource corresponds to the ID of the relevant resource
+    ///  in the storage.
+    /// @param tokenId The ID of the token of which to retrieve the active resource set
+    /// @return An array of uint64 resource IDs corresponding to active resources
+    function getActiveResources(uint256 tokenId) external view returns(uint64[] memory);
+
+    /// @notice Returns an array of uint64 identifiers from the pending resources
+    ///  array for resource lookup.
+    /// @dev Each uint64 resource corresponds to the ID of the relevant resource
+    ///  in the storage.
+    /// @param tokenId The ID of the token of which to retrieve the pending resource set
+    /// @return An array of uint64 resource IDs corresponding to pending resources
+    function getPendingResources(uint256 tokenId) external view returns(uint64[] memory);
+
+    /// @notice Returns an array of uint16 resource priorities.
+    /// @dev No validation is done on resource priority ranges, sorting MUST be
+    ///  handled by the frontend.
+    /// @param tokenId The ID of the token of which to retrieve the active resource set
+    ///  priorities
+    /// @return An array of uint16 resource priorities corresponding to active resources
+    function getActiveResourcePriorities(uint256 tokenId) external view returns(uint16[] memory);
+
+    /// @notice Returns the uint64 resource ID that would be overwritten when accepting the
+    ///  pending resource with ID resId on token.
+    /// @param tokenId The ID of the token of which we want to overwrite the pending resource
+    /// @param resId The resource ID which MAY overwrite another
+    /// @return A uint64 corresponding to the resource ID of the resource that would be overwritten
+    function getResourceOverwrites(uint256 tokenId, uint64 resId) external view returns(uint64);
+
+    /// @notice Returns the metadata of the resource associated with `resourceId`
+    /// @param resourceId The ID of the resource for which we are trying to retrieve the
+    ///  resource metadata
+    /// @return The metadata of the resource with ID equal to `resourceId`
+    function getResourceMeta(uint64 resourceId) external view returns (string memory);
```

Comment:
The output of this function isn't defined in the Specification section. Is it a URL? JSON?